neural_modelling/src/neuron/plasticity/stdp/synapse_dynamics_stdp_mad_impl.c

```diff
@@ -86,37 +98,43 @@ post_event_history_t *post_event_history;
 // Synapse update loop
 //---------------------------------------
 static inline final_state_t plasticity_update_synapse(
-        uint32_t time,
+        const uint32_t time,
         const uint32_t last_pre_time, const pre_trace_t last_pre_trace,
         const pre_trace_t new_pre_trace, const uint32_t delay_dendritic,
         const uint32_t delay_axonal, update_state_t current_state,
-        const post_event_history_t *post_event_history) {
+        const post_event_history_t *post_event_history,
+        const uint32_t is_autapse) {
+    use(is_autapse);
+
     // Apply axonal delay to time of last presynaptic spike
     const uint32_t delayed_last_pre_time = last_pre_time + delay_axonal;
 
     // Get the post-synaptic window of events to be processed
     const uint32_t window_begin_time =
             (delayed_last_pre_time >= delay_dendritic)
             ? (delayed_last_pre_time - delay_dendritic) : 0;
-    const uint32_t window_end_time = time + delay_axonal - delay_dendritic;
+    const uint32_t delayed_pre_time = time + delay_axonal;
+    const uint32_t window_end_time =
+            (delayed_pre_time >= delay_dendritic)
+            ? (delayed_pre_time - delay_dendritic) : 0;
     post_event_window_t post_window = post_events_get_window_delayed(
             post_event_history, window_begin_time, window_end_time);
 
     log_debug("\tPerforming deferred synapse update at time:%u", time);
-    log_debug("\t\tbegin_time:%u, end_time:%u - prev_time:%u, num_events:%u",
+    log_debug("\t\tbegin_time:%u, end_time:%u - prev_time:%u (valid %u), num_events:%u",
             window_begin_time, window_end_time, post_window.prev_time,
-            post_window.num_events);
+            post_window.prev_time_valid, post_window.num_events);
 
     // print_event_history(post_event_history);
     // print_delayed_window_events(post_event_history, window_begin_time,
     //		   window_end_time, delay_dendritic);
```

Comment:
It might be better to uncomment these lines and put them inside an if LOG_LEVEL >= LOG_DEBUG instead.

Reply:
Done.  This is the protection of the aforementioned method, so leaving it as log_info above makes more sense in this context.